lua/wikis/commons/Widget/Match/Page/MapVeto.lua

```diff
@@ -11,21 +11,47 @@ local Class = require('Module:Class')
 local Image = require('Module:Image')
 local Lua = require('Module:Lua')
 
+local Condition = Lua.import('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
 local Widget = Lua.import('Module:Widget')
 local HtmlWidgets = Lua.import('Module:Widget/Html/All')
 local WidgetUtil = Lua.import('Module:Widget/Util')
 local Link = Lua.import('Module:Widget/Basic/Link')
 
+---@alias VetoRound {name: string, link: string, type: 'pick'|'ban'|'decider', round: integer, by: standardOpponent}
+
 ---@class MatchPageMapVetoParameters
----@field vetoRounds {name: string, link: string, type: 'pick'|'ban'|'decider', round: integer, by: standardOpponent}[]
+---@field vetoRounds VetoRound[]
 
 ---@class MatchPageMapVeto: Widget
 ---@operator call(MatchPageMapVetoParameters): MatchPageMapVeto
 ---@field props MatchPageMapVetoParameters
 local MatchPageMapVeto = Class.new(Widget)
 
+---@private
+---@param map string
+---@return datapoint?
+function MatchPageMapVeto._getMapData(map)
+	local condition = ConditionTree(BooleanOperator.all)
+		:add(ConditionNode(ColumnName('type'), Comparator.eq, 'map'))
+		:add(ConditionNode(ColumnName('name'), Comparator.eq, map))
```

Comment:
why name instead of pagename?
couldn't we have cases where 2 maps have the same (display) name but different pagenames?